docs/import/list_ee_assets.js

```diff
@@ -1,50 +1,119 @@
 import {eeLegacyPathPrefix, legacyStateDir} from '../ee_paths.js';
 import {LayerType} from '../firebase_layers.js';
+import {convertEeObjectToPromise} from '../map_util.js';
 
 export {getDisasterAssetsFromEe, getStatesAssetsFromEe};
 
 /**
- * Requests all assets in ee directories corresponding to given states.
- * @param {Array<string>} states e.g. ['WA']
- * @return {Promise<Array<Array<string | Array<string>>>>} 2-d array of all
- *     retrieved assets in the form [['WA', {'asset/path': LayerType,...}], ...]
+ * Cache for the results of getStatesAssetsFromEe for each state.
+ * @type {Map<string, Promise<Map<string, string>>>}
+ */
+const stateAssetPromises = new Map();
+
+/**
+ * Gets all assets in ee directories corresponding to given states. Caches
+ * results. Marks assets with their type and whether or not they should be
+ * disabled when put into a select. Here, disabling any assets that aren't
+ * feature collections.
+ * @param {Array<string>} states
+ * @return {Map<string, Promise<Map<string, {disabled: boolean}>>>}
  */
 function getStatesAssetsFromEe(states) {
```

Comment:
Doesn't need to happen in this PR, but at this point it's not clear why we request multiple states like this. Would make more sense to just request each state individually. The original reason, of listing all state folders and creating them if necessary, is long gone.

Reply:
Are you imagining a promise per states instead?

Reply:
promise per individual state, yes. Aren't we basically already doing that here?